src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -510,4 +511,22 @@ close(int dev_handle) const
   pcidev::get_dev(get_device_id(), false)->close(dev_handle);
 }
 
+void
+device_linux::
+load_xclbin(std::vector<char> &buffer) const {
+  int ret = 0;
+
+  try {
```

Comment:
Magic "binary value" check (e.g., "xclbin2" should be done here.  This this check should be a BINARY check and NOT text.

Reply:
Moved. I'm not sure how to do a binary check against a string...

Reply:
Then don't use a string :^).  The underlying issue is that "by chance" the magic xclbin image value is "xclbin2", which is all ascii, imaging instead it is "0x78636c62696e32"  How would you validate these value?

Fortunately, there are memory functions that can help (e.g., std::memcmp).

**_Note_**
This code does work in this case, but it is something that I would recommend against.